RPS

bbiehl opened the pull request.

bbiehl committed:
Phase 1 - Build RPS, Part 1

bbiehl commented:
I works on repl.it, can't get it to do anything when trying to test out from the iTerm command line.

bbiehl committed:
Phase 1 - Build RPS, part 2

bbiehl committed:
Phase 2

bbiehl committed:
Phase 2 expanded, not working

bbiehl commented:
Tried to expand Phase 2 to include how the player won, i.e "Rock Smashes Scissors", "Paper Covers Rock", or "Scissors Cut Paper".  I am going to revert to my working code in order to get started working on Phase 3.  But would like to come back to this non-working Phase 2 at a later time.

halfghaninne commented:
Can you tell me a little more about what you expect to get when you run the code in this latest commit and what error you are getting instead? I think I have an idea of what the issue is.

bbiehl commented:
When we run the code we should expect to not only expect which player won to be returned, but how they won i.e. "Rock Smashes Scissors", "Paper Covers Rock", or "Scissors Cut Paper".  I think I know how to make it work now after resting up and feeling normal again.  Will let you know if I run into any roadblocks.

bbiehl committed:
Phase 2 Expanded - not working

halfghaninne commented:
Okay, great! Let me know if you have questions about resolving that issue! My inclination is to think of the kind of Objects that the `how_won` method is returning.

bbiehl committed:
Phase 2, getting closer

bbiehl commented:
When we induce a "draw" we get the following error.

Player 1, unleash your weapon:
rock
Player 2, unleash your weapon:
rock
lib/rock-paper-scissors.rb:57:in `display_how_game_won': undefined local variable or method `p1_weapon' for main:Object (NameError)
	from lib/rock-paper-scissors.rb:76:in `<main>'

This method is does not know the values for <code>p1_weapon</code> and <code>p2_weapon</code>
<code>def display_how_game_won(x_ray)</code>
<code>  if x_ray == 1</code>
<code>    puts "Paper Covers Rock"</code>
<code>  elsif x_ray == 2</code>
<code>    puts "Scissors Cut Paper"</code>
<code>  elsif x_ray == 3</code>
<code>    puts "Rock Smashes Scissors"</code>
<code>  else x_ray == 0</code>
<code>    puts "#{p1_weapon} equals #{p2_weapon}"</code>
<code>  end</code>
<code>end</code>

bbiehl committed:
Phase 2, single game

bbiehl commented:
Refactored and able to play a single game.

bbiehl committed:
Phase 2 refactor and TomDoc

bbiehl commented:
Still only able to play a single game, but now TomDoc'd.

bbiehl committed:
Phase 2 refactor and TomDoc

bbiehl commented:
Improved TomDoc.

bbiehl committed:
Testing

bbiehl committed:
Testing

bbiehl commented:
First attempt at testing. 

  1) Error:
RockPaperScissorsTests#test_rock_vs_paper:
ArgumentError: wrong number of arguments (given 2, expected 3)
    /Users/bradleybiehl/Code/01-29-Rock-Paper-Scissors/ruby-app__rock-paper-scissors/lib/rock-paper-scissors.rb:132:in `display_how_game_won'
    /Users/bradleybiehl/Code/01-29-Rock-Paper-Scissors/ruby-app__rock-paper-scissors/tests/rbs_test.rb:8:in `test_rock_vs_paper'

bbiehl committed:
Testing

bbiehl committed:
Testing

bbiehl committed:
Testing

bbiehl commented:
I'd like to join the queue tomorrow morning to go over this.  The test runs, but is not going in-depth enough.

bbiehl committed:
OOP

bbiehl commented:
OOP added

bbiehl committed:
RPS-OOP full test

bbiehl commented:
### Trying to run rps_oop.rb

lib/rps_oop.rb:70: formal argument cannot be an instance variable
  def display_game_result_how_won(w, @p1_weapon, @p2_weapon)
                                               ^
lib/rps_oop.rb:70: formal argument cannot be an instance variable
  def display_game_result_how_won(w, @p1_weapon, @p2_weapon)

#### from line 70
#### formal argument cannot be an instance variable
#### Action: changed @p1_weapon to weapon_p1, and @p2_weapon to weapon_p2
#### Status: Fixed

-------

### rake unsuccessful

  1) Error:
RockPaperScissorsOOPTests#test_rock_vs_rock:
NoMethodError: undefined method `set_p1_weapon' for #<Game:0x007ff240b07b58>
Did you mean?  get_p1_weapon
               get_p2_weapon
    /Users/bradleybiehl/Code/01-29-Rock-Paper-Scissors/ruby-app__rock-paper-scissors/tests/rps_oop_test.rb:7:in `test_rock_vs_rock'

#### from rps_oop_test.rb line 7
#### no method error
#### Action: changed set_pl_weapon to get_p1_weapon
#### Action: changed set_p2_weapon to get_p2_weapon
#### Status: Fixed

### rake successful
#### 4 runs, 14 assertions, 0 failures, 0 errors, 0 skips

-------

### rake unsuccessful

 1) Error:
RockPaperScissorsOOPTests#test_rock_vs_scissors:
NameError: undefined local variable or method `p1_weapon' for #<Game:0x007fd4209e5100 @p1_weapon="rock", @p2_weapon="scissors">
Did you mean?  @p1_weapon
               @p2_weapon
    /Users/bradleybiehl/Code/01-29-Rock-Paper-Scissors/ruby-app__rock-paper-scissors/lib/rps_oop.rb:45:in `how_game_was_won'
    /Users/bradleybiehl/Code/01-29-Rock-Paper-Scissors/ruby-app__rock-paper-scissors/tests/rps_oop_test.rb:37:in `test_rock_vs_scissors'

#### rps_oop.rb line 45 'how_game_was_won'
<code>elsif (p1_weapon == "paper" && p2_weapon == "scissors") || (p1_weapon == "scissors" && p2_weapon == "paper")</code>

#### rps_oop_test.rb line 37
<code>assert_equal("rss", sample_game.how_game_was_won)</code>

#### syntax error
#### Action: modified rps_oop.rb 'how_game_was_won' method to reflect instance variables
### rake successful

----
### Full Test of every scenario
#### 12 runs, 30 assertions, 0 failures, 0 errors, 0 skips

sumeetjain commented:
:+1:

bbiehl committed:
Stashing this for later.  Going to try some new classes.

bbiehl committed:
rogue commit!!!
